### ISSUE TITLE
When idle, extension kills Panel (pitch black) and upper half of screen

### ISSUE THREAD

**Issue by zilexa** [G-dH/custom-hot-corners-extended#17]
On Manjaro Gnome (on a HP x360 Spectre 14 laptop Intel i5-1035G4. No Nvidia so default is Wayland) with latest updates, I have this strange issue. I only installed your extension and https://github.com/harshadgavali/gnome-gesture-improvements and did not change the built in extensions, only disabled X11 gestures.

To reproduce: 
1. reboot and login. Open a window for example Firefox, fullscreen
2. Wait about 15-30min
3. Notice firefox suddenly only covers the bottom half of the screen, the wallpaper is visible on the upper half.
4. Notice also Panel is now an empty black bar. 

I cannot do much, I usually use CTRL+ALT+DEL to power off and reboot. 

When I disable the extension, I get the same issue strangely. 

I am not sure how to produce the right logs to see what is happening here (since I need to reboot to be able to do anything).

**Comment by G-dH** [comment]
If I understand correctly, the issue persist even if you disable my extension? And if you uninstall it and restart the GNOME Shell? When idle, the extension itself does nothing but checking the corners every 3 seconds, if this feature is enabled.
Try to look into the log if there is something directly related to the CHC-E:

    journalctl /usr/bin/gnome-shell --since "2 days ago" | grep corners-extended

**Comment by zilexa** [comment]
Sorry I meant: the moment I hit the toggle to disable the extension, it happens. After a reboot I don't have this issue anymore (and the extension is still disabled).

I left it disabled the rest of the day and had no issues.
I will search through the logfile.

**Comment by G-dH** [comment]
What built-in extensions are enabled?

**Comment by zilexa** [comment]
```
gsettings get org.gnome.shell enabled-extensions                                                                                                                ✔ 
['pamac-updates@manjaro.org', 'gnome-ui-tune@itstime.tech', 'ding@rastersoft.com', 'appindicatorsupport@rgcjonas.gmail.com', 'dash-to-panel@jderose9.github.com', 'arcmenu@arcmenu.com', 'gestureImprovements@gestures']
```
Note the only built-in extension I disabled is X11 Gestures. 
I will now enable Custom Hot Corners Extended and report log as soon as it happens again.

**Comment by G-dH** [comment]
Does the issue occur every time you disable Custom Hot Corners - Extended?

**Comment by zilexa** [comment]
here we go: 
1. enable the extension
2. reboot, login, open a window
3. don't touch the laptop, wait until screen goes dark (in my case via Settings > Power > Dim screen set to 8 minutes)
4. wake up the screen by touching touchpad or keyboard
5. Notice the panel is gone, instead that area is pitch black. Also, I can hit ALT to see the Firefox menu bar, but I cannot click any of the menus. Nothing happens. 

```

-- Journal begins at Tue 2021-11-02 21:46:36 CET, ends at Mon 2021-11-15 11:41:03 CET. --
nov 15 11:25:38 Idefix gnome-shell[1044]: Adding device '/dev/dri/card0' (i915) using atomic mode setting.
nov 15 11:25:38 Idefix gnome-shell[1044]: Boot VGA GPU /dev/dri/card0 selected as primary
nov 15 11:25:39 Idefix gnome-shell[1044]: libinput error: client bug: timer event10 hold: scheduled expiry is in the past (-110ms), your system is too slow
nov 15 11:25:39 Idefix gnome-shell[1044]: Using public X11 display :0, (using :1 for managed services)
nov 15 11:25:39 Idefix gnome-shell[1044]: Using Wayland display name 'wayland-0'
nov 15 11:25:39 Idefix gnome-shell[1044]: Skipping parental controls support as it’s disabled
nov 15 11:25:39 Idefix gnome-shell[1044]: Unset XDG_SESSION_ID, getCurrentSessionProxy() called outside a user session. Asking logind directly.
nov 15 11:25:39 Idefix gnome-shell[1044]: Will monitor session 3
nov 15 11:25:39 Idefix gnome-shell[1044]: Telepathy is not available, chat integration will be disabled.
nov 15 11:25:40 Idefix gnome-shell[1044]: Some code accessed the property 'SecondaryMonitorDisplay' on the module 'workspacesView'. That property was defined with 'let' or 'const' inside >
nov 15 11:25:40 Idefix gnome-shell[1044]: Some code accessed the property 'ThumbnailsBox' on the module 'workspacesView'. That property was defined with 'let' or 'const' inside the module>
nov 15 11:25:40 Idefix gnome-shell[1044]: st_widget_get_theme_node called on the widget [0x557a84997290 StBin:insensitive ("")] which is not in the stage.
nov 15 11:25:40 Idefix gnome-shell[1044]: st_widget_get_theme_node called on the widget [0x557a84996a80 Gjs_ui_panel_AppMenuButton.panel-button:insensitive ("")] which is not in the stage.
nov 15 11:25:40 Idefix gnome-shell[1044]: st_widget_get_theme_node called on the widget [0x557a84998540 StBin:insensitive "appMenu" ("")] which is not in the stage.
nov 15 11:25:40 Idefix gnome-shell[1044]: st_widget_get_theme_node called on the widget [0x557a84999d20 StBoxLayout.panel-status-menu-box:insensitive ("")] which is not in the stage.
nov 15 11:25:40 Idefix gnome-shell[1044]: st_widget_get_theme_node called on the widget [0x557a8499faf0 Gjs_ui_animation_Spinner:insensitive] which is not in the stage.
nov 15 11:25:40 Idefix gnome-shell[1044]: st_widget_get_theme_node called on the widget [0x557a84997290 StBin:insensitive ("")] which is not in the stage.
nov 15 11:25:40 Idefix gnome-shell[1044]: st_widget_get_theme_node called on the widget [0x557a84996a80 Gjs_ui_panel_AppMenuButton.panel-button:insensitive ("")] which is not in the stage.
nov 15 11:25:40 Idefix gnome-shell[1044]: st_widget_get_theme_node called on the widget [0x557a84998540 StBin:insensitive "appMenu" ("")] which is not in the stage.
nov 15 11:25:40 Idefix gnome-shell[1044]: st_widget_get_theme_node called on the widget [0x557a84999d20 StBoxLayout.panel-status-menu-box:insensitive ("")] which is not in the stage.
nov 15 11:25:40 Idefix gnome-shell[1044]: st_widget_get_theme_node called on the widget [0x557a8499faf0 Gjs_ui_animation_Spinner:insensitive] which is not in the stage.
nov 15 11:25:40 Idefix gnome-shell[1044]: Error looking up permission: GDBus.Error:org.freedesktop.portal.Error.NotFound: No entry for geolocation
nov 15 11:25:41 Idefix gnome-shell[1044]: ATK Bridge is disabled but a11y has already been enabled.
nov 15 11:25:41 Idefix gnome-shell[1044]: GNOME Shell started at Mon Nov 15 2021 11:25:40 GMT+0100 (Central European Standard Time)
nov 15 11:25:41 Idefix gnome-shell[1044]: Registering session with GDM
nov 15 11:25:41 Idefix gnome-shell[599]: Connection to xwayland lost
nov 15 11:26:10 Idefix gnome-shell[1044]: updates_checker.vala:71: check updates
nov 15 11:26:11 Idefix gnome-shell[1044]: updates_checker.vala:101: 0 updates found
nov 15 11:33:59 Idefix gnome-shell[1044]: Timelines with detached actors are not supported
nov 15 11:33:59 Idefix gnome-shell[1044]: Timelines with detached actors are not supported
nov 15 11:33:59 Idefix gnome-shell[1044]: Timelines with detached actors are not supported
nov 15 11:33:59 Idefix gnome-shell[1044]: [Custom Hot Corners - Extended] extension suspended
nov 15 11:34:00 Idefix gnome-shell[1044]: DING: Gjs-Message: 11:25:42.675: JS LOG: Error connecting to Nautilus
                                          GNOME nautilus 40.3
nov 15 11:36:27 Idefix gnome-shell[1044]: Timelines with detached actors are not supported
nov 15 11:36:27 Idefix gnome-shell[1044]: Timelines with detached actors are not supported
nov 15 11:36:27 Idefix gnome-shell[1044]: Timelines with detached actors are not supported
nov 15 11:36:27 Idefix gnome-shell[1044]: JS ERROR: Extension dash-to-panel@jderose9.github.com: TypeError: array is undefined
                                          findIndex@/usr/share/gnome-shell/extensions/dash-to-panel@jderose9.github.com/utils.js:317:9
                                          find@/usr/share/gnome-shell/extensions/dash-to-panel@jderose9.github.com/utils.js:330:26
                                          newUpdateHotCorners@/usr/share/gnome-shell/extensions/dash-to-panel@jderose9.github.com/panelManager.js:640:27
                                          _setPanelPosition@/usr/share/gnome-shell/extensions/dash-to-panel@jderose9.github.com/panel.js:1049:28
                                          enable@/usr/share/gnome-shell/extensions/dash-to-panel@jderose9.github.com/panel.js:231:14
                                          _createPanel@/usr/share/gnome-shell/extensions/dash-to-panel@jderose9.github.com/panelManager.js:405:15
                                          enable@/usr/share/gnome-shell/extensions/dash-to-panel@jderose9.github.com/panelManager.js:84:34
                                          _enable@/usr/share/gnome-shell/extensions/dash-to-panel@jderose9.github.com/extension.js:104:18
                                          enable@/usr/share/gnome-shell/extensions/dash-to-panel@jderose9.github.com/extension.js:66:5
lines 1-48...skipping...
-- Journal begins at Tue 2021-11-02 21:46:36 CET, ends at Mon 2021-11-15 11:41:03 CET. --
nov 15 11:25:38 Idefix gnome-shell[1044]: Adding device '/dev/dri/card0' (i915) using atomic mode setting.
nov 15 11:25:38 Idefix gnome-shell[1044]: Boot VGA GPU /dev/dri/card0 selected as primary
nov 15 11:25:39 Idefix gnome-shell[1044]: libinput error: client bug: timer event10 hold: scheduled expiry is in the past (-110ms), your system is too slow
nov 15 11:25:39 Idefix gnome-shell[1044]: Using public X11 display :0, (using :1 for managed services)
nov 15 11:25:39 Idefix gnome-shell[1044]: Using Wayland display name 'wayland-0'
nov 15 11:25:39 Idefix gnome-shell[1044]: Skipping parental controls support as it’s disabled
nov 15 11:25:39 Idefix gnome-shell[1044]: Unset XDG_SESSION_ID, getCurrentSessionProxy() called outside a user session. Asking logind directly.
nov 15 11:25:39 Idefix gnome-shell[1044]: Will monitor session 3
nov 15 11:25:39 Idefix gnome-shell[1044]: Telepathy is not available, chat integration will be disabled.
nov 15 11:25:40 Idefix gnome-shell[1044]: Some code accessed the property 'SecondaryMonitorDisplay' on the module 'workspacesView'. That property was defined with 'let' or 'const' inside >
nov 15 11:25:40 Idefix gnome-shell[1044]: Some code accessed the property 'ThumbnailsBox' on the module 'workspacesView'. That property was defined with 'let' or 'const' inside the module>
nov 15 11:25:40 Idefix gnome-shell[1044]: st_widget_get_theme_node called on the widget [0x557a84997290 StBin:insensitive ("")] which is not in the stage.
nov 15 11:25:40 Idefix gnome-shell[1044]: st_widget_get_theme_node called on the widget [0x557a84996a80 Gjs_ui_panel_AppMenuButton.panel-button:insensitive ("")] which is not in the stage.
nov 15 11:25:40 Idefix gnome-shell[1044]: st_widget_get_theme_node called on the widget [0x557a84998540 StBin:insensitive "appMenu" ("")] which is not in the stage.
nov 15 11:25:40 Idefix gnome-shell[1044]: st_widget_get_theme_node called on the widget [0x557a84999d20 StBoxLayout.panel-status-menu-box:insensitive ("")] which is not in the stage.
nov 15 11:25:40 Idefix gnome-shell[1044]: st_widget_get_theme_node called on the widget [0x557a8499faf0 Gjs_ui_animation_Spinner:insensitive] which is not in the stage.
nov 15 11:25:40 Idefix gnome-shell[1044]: st_widget_get_theme_node called on the widget [0x557a84997290 StBin:insensitive ("")] which is not in the stage.
nov 15 11:25:40 Idefix gnome-shell[1044]: st_widget_get_theme_node called on the widget [0x557a84996a80 Gjs_ui_panel_AppMenuButton.panel-button:insensitive ("")] which is not in the stage.
nov 15 11:25:40 Idefix gnome-shell[1044]: st_widget_get_theme_node called on the widget [0x557a84998540 StBin:insensitive "appMenu" ("")] which is not in the stage.
nov 15 11:25:40 Idefix gnome-shell[1044]: st_widget_get_theme_node called on the widget [0x557a84999d20 StBoxLayout.panel-status-menu-box:insensitive ("")] which is not in the stage.
nov 15 11:25:40 Idefix gnome-shell[1044]: st_widget_get_theme_node called on the widget [0x557a8499faf0 Gjs_ui_animation_Spinner:insensitive] which is not in the stage.
nov 15 11:25:40 Idefix gnome-shell[1044]: Error looking up permission: GDBus.Error:org.freedesktop.portal.Error.NotFound: No entry for geolocation
nov 15 11:25:41 Idefix gnome-shell[1044]: ATK Bridge is disabled but a11y has already been enabled.
nov 15 11:25:41 Idefix gnome-shell[1044]: GNOME Shell started at Mon Nov 15 2021 11:25:40 GMT+0100 (Central European Standard Time)
nov 15 11:25:41 Idefix gnome-shell[1044]: Registering session with GDM
nov 15 11:25:41 Idefix gnome-shell[599]: Connection to xwayland lost
nov 15 11:26:10 Idefix gnome-shell[1044]: updates_checker.vala:71: check updates
nov 15 11:26:11 Idefix gnome-shell[1044]: updates_checker.vala:101: 0 updates found
nov 15 11:33:59 Idefix gnome-shell[1044]: Timelines with detached actors are not supported
nov 15 11:33:59 Idefix gnome-shell[1044]: Timelines with detached actors are not supported
nov 15 11:33:59 Idefix gnome-shell[1044]: Timelines with detached actors are not supported
nov 15 11:33:59 Idefix gnome-shell[1044]: [Custom Hot Corners - Extended] extension suspended
nov 15 11:34:00 Idefix gnome-shell[1044]: DING: Gjs-Message: 11:25:42.675: JS LOG: Error connecting to Nautilus
                                          GNOME nautilus 40.3
nov 15 11:36:27 Idefix gnome-shell[1044]: Timelines with detached actors are not supported
nov 15 11:36:27 Idefix gnome-shell[1044]: Timelines with detached actors are not supported
nov 15 11:36:27 Idefix gnome-shell[1044]: Timelines with detached actors are not supported
nov 15 11:36:27 Idefix gnome-shell[1044]: JS ERROR: Extension dash-to-panel@jderose9.github.com: TypeError: array is undefined
                                          findIndex@/usr/share/gnome-shell/extensions/dash-to-panel@jderose9.github.com/utils.js:317:9
                                          find@/usr/share/gnome-shell/extensions/dash-to-panel@jderose9.github.com/utils.js:330:26
                                          newUpdateHotCorners@/usr/share/gnome-shell/extensions/dash-to-panel@jderose9.github.com/panelManager.js:640:27
                                          _setPanelPosition@/usr/share/gnome-shell/extensions/dash-to-panel@jderose9.github.com/panel.js:1049:28
                                          enable@/usr/share/gnome-shell/extensions/dash-to-panel@jderose9.github.com/panel.js:231:14
                                          _createPanel@/usr/share/gnome-shell/extensions/dash-to-panel@jderose9.github.com/panelManager.js:405:15
                                          enable@/usr/share/gnome-shell/extensions/dash-to-panel@jderose9.github.com/panelManager.js:84:34
                                          _enable@/usr/share/gnome-shell/extensions/dash-to-panel@jderose9.github.com/extension.js:104:18
                                          enable@/usr/share/gnome-shell/extensions/dash-to-panel@jderose9.github.com/extension.js:66:5
                                          _callExtensionEnable@resource:///org/gnome/shell/ui/extensionSystem.js:167:32
```

Will attach a photo as well.

**Comment by zilexa** [comment]
Fyi a screenshot of the situation before 'it'  happens: 

![Screenshot from 2021-11-15 11-59-18](https://user-images.githubusercontent.com/3430004/141770569-84c2caad-c91c-462b-a87a-9f7aa1cd2efc.png)

**Comment by G-dH** [comment]
I assume that if you disable the Dash to Panel extension, the issue will not occur too...

**Comment by zilexa** [comment]
Correct. 
Should I report this to the Dash to Panel developer?
I am not sure where this can be fixed.

**Comment by G-dH** [comment]
I found the trigger - ArcMenu inside the Dash to Panel, set to modify hot corner behavior. It's a conflict between CHC-E and ArcMenu, DtP is a collateral damage. It's an ArcMenu fault that it doesn't handle the exception, but it's still impossible to use ArcMenu set to modify the hot corner because CHC-E will overwrite that corner anyway. The only solution in this case would be to trigger the ArcMenu from CHC-E, that's what we've discussed before.

**Comment by G-dH** [comment]
I contacted the ArcMenu developer and he added support for external menu activation so now I can add support for ArcMenu directly to CHC-E. I'll let you know when it's ready.

**Comment by zilexa** [comment]
That is fantastic! Thanks

**Comment by G-dH** [comment]
The code is already available in our repositories, for the ArcMenu it's enough to update the `controller.js' file which you can find here: https://gitlab.com/arcmenu/ArcMenu.

**Comment by G-dH** [comment]
CHC-E with the ArcMenu support is already on e.g.o.

**Comment by zilexa** [comment]
Thanks! What does e.g.o mean?
Sorry, I would just wait until the updates come in of ArcMenu and CHC-E. I don't want to mess too much with individual files..

**Comment by G-dH** [comment]
e.g.o == extensions.gnome.org, so your CHC-E should update automatically.

**Comment by zilexa** [comment]
Finally had the chance to test this properly. Unfortunately I believe there still is something funky happening. 
I did a clean install of Manjaro Gnome (disk wipe). 

1. I replaced the controller.js file.
2. Then got update notification: lot's of updates available, including for Arcmenu. I updated the system and rebooted. 
3. Installed CHC-E. 
4. Checked if "Modify Hot Corner.." in Extensions>ArcMenu was disabled. It is disabled by default. I did not touch it. 
5. Checked wihat Top Left did: the Gnome/ArcMenu default action happened (show activities). 

Question: Are you sure I should not _disable_ the hotcorner action in ArcMenu > 'Modify hot corner' ? 
I continued my tests without touching this setting as it is disabled by default.

Now testing CHC-E:
6. In CHC-E, I enabled the top-left corner for ArMenu. 
7. Tested the top left hot corner: indeed now ArcMenu shows
--> problem 1: the mouse gets stuck in this corner and I see the greeny animation that the corner has been touched. I can release the mouse after moving more. 
--> minor problem 2: ArcMenu stays open. It doesn't close if you move away. 
8. Rebooted again. 
9. Tested top left: 
--> Main problem 2: it's back to the default Show Activities. CHC-E has no effect. 

How can I help here?

**Comment by G-dH** [comment]
You're right, this issue persists and can be triggered on Manjaro by disabling/enabling all of the extensions at once (that in fact happens, when you disable any extension after login, all other extensions are re-enabled). I wasn't able to reproduce the issue on Ubuntu or Fedora.
CHC-E can open the ArcMenu but cannot close it because the menu is modal and grabs all mouse events so CHC-E can't react on them.
I'll see what I can do, but how the ArcMenu and Dash to Dock behave is not up to me.

**Comment by AndrewZaech** [comment]
Hi @G-dH 

I did some testing on a Manjaro VM - https://gitlab.com/arcmenu/ArcMenu/-/issues/154#note_745026487

At the start of each test I reloaded GNOME shell with Alt+F2 'r' to get a fresh start.

Can you try testing the scenarios listed on that post?

**Comment by zilexa** [comment]
I'll try tomorrow, I will be on holiday 2 wks after that, no laptop 🍶

**Comment by G-dH** [comment]
Hi @AndrewZaech, I tested all the  combinations before and the result was that only the combination of all 3 extensions caused the issue. I'll test it again of course, soon..

**Comment by AndrewZaech** [comment]
I removed the 'Modify Activities Hot Corner' feature from ArcMenu. It was a bit out of scope for ArcMenu to have such a feature in the first place. 

Hopefully now there is no longer any conflict between our extensions.

If you do any further testing make sure to build ArcMenu from master branch

**Comment by zilexa** [comment]
FYI, with Manjaro Gnome on non-Nvidia systems, Wayland is default. Perhaps that makes a difference? I am on Wayland. For example `Alt + F2 + r` or restart is not available on Gnome.

**Comment by G-dH** [comment]
@AndrewZaech : Doesn't help, the conflict is actually between DtP and CHC-E, I don't know, how I could miss this during those first tests, but ArcMenu can be disabled and the issue persists. When extensions are enabled Dash to Panel stores the original `upadateHotCorners()` function and replaces it by its own function. Then starts CHC-E (it's start is delayed to avoid corner overrides from other extensions at session start and the screen unlock and to avoid blocking the screen unlock animation by registering keyboard shortcuts) and CHC-E does the same, except that this time CHC-E stores the replaced function. That's the reason I recently disabled calling this function in `disable()` after I replace it with the "original" one, to avoid calling dead DtP. But in Manjaro something else happens and the conflict crashes the session.

It is enough to deactivate Hot Corner in GNOME Tweaks to prevent DtP to touch the corner and conflict will not happen.
I'll add the corner disabling into the CHC-E enable() function which should prevent such conflicts. It used to be there, but I removed it long time ago, not sure why...

And I don't think that the `Modify Activities Hot Corner` feature is out of scope, for ArcMenu users that don't need CHC-E. Although I didn't expect that kind of injection into the existing corners :) CHC-E is able to override the corner modification within 3 seconds, there shouldn't be any conflict.

**Comment by zilexa** [comment]
Strangely, when I:
- Arcmenu enabled, DtP enabled, CHC-E enabled with Arcmenu set to trigger top-left, `Modify Acitivities Hot Corner` disabled in Arc Menu AND in Tweaks --> I still get Activities Overview and sometimes also Arc Menu opens.. 

It seems, disabling the default action via Tweaks does not have the desired effect. I will logout-login to see if that helps.

**Comment by AndrewZaech** [comment]
@G-dH: Thank you for the detailed explanation. Glad to see the underlying issue has been found!

I will revert the recent commit that removed `Modify Activities Hot Corner` from ArcMenu based on the above feedback

**Comment by G-dH** [comment]
@zilexa: I'm unable to reproduce the stuck mouse issue on my Manjaro installation.

**Comment by zilexa** [comment]
@G-dH After logout/login, I do not have the stuck mouse issue anymore :) 

Thanks a lot for the dedication, it all works fine now. 

If I may ask a totally offtopic question:
What would be the 'recommended' way to install your extension automatically (via post install bash script) in such a way that the distro will keep it up to date? 
Untill now I used the zip file. 
But I found the extension via  Add/Remove Programs> enable Aur > search for it > and install it.
This would be similar to `sudo pamac install --no-confirm gnome-shell-extension-custom-hot-corners-extended`. 
Just wondering if it is OK to rely on the AUR version to stay up to date (if I don't want to use browser extension).
